### PR TITLE
Fix errors in deleting routes and route rules

### PR DESCRIPTION
**Pull request by karthiksundaravel**
During the reapply of nmstate templates, the mismatch in routes and route rules will result in deletion of some routes and route rules. The deletion of some of the routes conflicts with the addition of new entries. In order to avoid the conflict, when there is a difference between the existing route/route rules and the desired ones, the existing routes and route rules were flushed and the new entries were made.

Linked Jira: https://issues.redhat.com/browse/OSPRH-9904

### CODE REVIEW

File: os_net_config/impl_nmstate.py [hunk 1]
```diff
@@ -471,7 +476,7 @@ def set_ifaces(self, iface_data):
         :param verify: boolean that determines if config will be verified
```
Comment (vcandapp) [inline os_net_config/impl_nmstate.py:476]: There is no verify input param

File: os_net_config/impl_nmstate.py [hunk 2]
```diff
@@ -471,7 +476,7 @@ def set_ifaces(self, iface_data):
...
         """
         state = {Interface.KEY: iface_data}
-        self.__dump_config(state, msg=f"Overall interface config")
+        self.__dump_config(state, msg=f"Prepared interface config")
```
Comment (vcandapp) [inline os_net_config/impl_nmstate.py:479]: Comments needs to be updated for:
   - rules_state (no input "name")
   - set_ifaces (no input "verify", return "state"
   - set_dns

File: os_net_config/impl_nmstate.py [hunk 3]
```diff
@@ -505,70 +510,69 @@ def set_rules(self, rule_data,):
         """
 
         state = {NMRouteRule.KEY: {NMRouteRule.CONFIG: rule_data}}
-        self.__dump_config(state, msg=f'Overall rules')
+        self.__dump_config(state, msg=f'Prepared rules are')
         return state
 
     def nmstate_apply(self, new_state, verify=True):
-        self.__dump_config(new_state, msg=f'Applying the config with nmstate')
+        self.__dump_key_config(new_state,
```
Comment (vcandapp) [inline os_net_config/impl_nmstate.py:517]: Need description for module params / return

**Review by vcandapp — CHANGES_REQUESTED** [review]

File: os_net_config/impl_nmstate.py (hunk 2 above)
Comment (karthiksundaravel) [reply os_net_config/impl_nmstate.py:478]: Done

File: os_net_config/impl_nmstate.py (hunk 1 above)
Comment (karthiksundaravel) [reply os_net_config/impl_nmstate.py]: done

**Comment by karthiksundaravel** [comment]
recheck

**Review by vcandapp — APPROVED** [review]
lgtm

**Comment by karthiksundaravel** [comment]
recheck

**Comment by dsneddon** [comment]
recheck

**Comment by karthiksundaravel** [comment]
recheck

**Pull request merged**